src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -26,23 +26,19 @@ public AuthenticationHandler(bool preAuthenticate, ICredentials credentials, Htt
             _innerHandler = innerHandler;
         }
 
-        private bool TrySetBasicAuthToken(HttpRequestMessage request)
-        {
-            NetworkCredential credential = _credentials.GetCredential(request.RequestUri, Basic);
-            if (credential == null)
-            {
-                return false;
-            }
-
-            request.Headers.Authorization = new AuthenticationHeaderValue(Basic, BasicAuthenticationHelper.GetBasicTokenForCredential(credential));
-            return true;
-        }
-
         protected internal override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
             if (_preAuthenticate)
             {
-                TrySetBasicAuthToken(request);
+                // Try using previous digest response WWWAuthenticate header
+                if (_digestResponse != null)
+                {
+                    AuthenticationHelper.TrySetDigestAuthToken(request, _credentials, _digestResponse);
+                }
+                else
+                {
+                    AuthenticationHelper.TrySetBasicAuthToken(request, _credentials);
```

Comment:
This is really bad from a security perspective. Basic auth sends the password in effectively plain text. If pre-authentication has been requested but this is the first request, the client will send the username in plain text. This means if I communicate with a server that I know uses digest auth and I care about not sending my password, I have to make sure that PreAuthenticate is disabled.